### PR TITLE
 Add a filter for primary publishing organisation

### PR DESCRIPTION
**Pull request by nacnudus**
Organisations often ask GDS content designers for a list of pages that they have
published.

Perhaps users also need checkboxes to search among all organisations (not just
the primary one) and to include descendant organisations.  But this is a start.

### CODE REVIEW

**Comment by nacnudus** [comment]
Question to users: Do you ever need to include non-primary publishing organisations, or organisations that are part of a larger one?

Replies:

- I'd say yes to organisations that are part of a larger one, like justice agencies
- The primary/secondary publishing organisation is just a Whitehall distinction. So if you're searching across all the apps, I find it helpful to search for all publishing orgs.

File: bigquery.ts [hunk 1]
```diff
@@ -198,30 +206,43 @@ const sendSearchQuery: SendSearchQuerySignature = async function(searchParams) {
...
-
-  const bqMainResults = results[0];
-  const bqMetaResults = results.length > 1 ? results[1] : [];
 
+  let bqMetaResults: MetaResult[] = [], results: any, bqMainResults: MainResult[] = [];
+  switch (searchParams.searchType) {
+    case SearchType.Taxon:
+      console.log('SearchType taxon');
+      results = await Promise.all(queries);
+      bqMainResults = results[0];
```
Comment (nacnudus) [inline bigquery.ts:215]: These lines can come before the `switch`, similarly lines 220-221

File: bigquery.ts [hunk 2]
```diff
@@ -198,30 +206,43 @@ const sendSearchQuery: SendSearchQuerySignature = async function(searchParams) {
...
+        selectedWordsWithoutQuotes.includes(' ')) {
+        queries.push(bigQuery(
+        `SELECT *
+         FROM search.thing
+         WHERE CONTAINS_SUBSTR(name, @selected_words_without_quotes)
+         ;`
+        , { selectedWordsWithoutQuotes }))
+      }
+      results = await Promise.all(queries);
+      bqMainResults = results[0]
```
Comment (nacnudus) [inline bigquery.ts:237]: This seems redundant, given what has happened above.

File: cypress/e2e/organisation-search.cy.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    cy.get('button#search').click();
+    cy.get('#results-heading', { timeout: 60000 }).contains('No results');
+    cy.get('ul#show-fields').should('not.exist');
+    cy.get('table#results-table').should('not.exist');
+  });
+
+  it('returns results for an existing organisation search', () => {
+    cy.visit('');
+    cy.contains('button', 'Organisations').click();
+    cy.get('input#organisation').type('Environment Agency');
```
Comment (nacnudus) [inline cypress/e2e/organisation-search.cy.ts:24]: I'd choose the Treasury, for longevity.

File: bigquery.ts (hunk 1 above)
Comment (maxf) [reply bigquery.ts:167]: No, because of the `default:` section where the meta search is added to `queries`

Comment (nacnudus) [reply bigquery.ts:167]: Can they come after, then?

File: bigquery.ts [hunk 4]
```diff
@@ -80,139 +86,113 @@ const sendInitQuery: SendInitQuerySignature = async function() {
...
-    homepage: bqTaxon[0].homepage,
-    description: bqTaxon[0].description,
-    level: parseInt(bqTaxon[0].level),
-    ancestorTaxons: bqTaxon[0].ancestorTaxons,
-    childTaxons: bqTaxon[0].childTaxons
-  };
+const getTaxonInfo = async function(name: string): Promise<Taxon[]> {
+  return await bigQuery(
+    `SELECT "Taxon" as type, * FROM search.taxon WHERE lower(name) = lower(@name);`, { name }
+  );
```
Comment (nacnudus) [inline bigquery.ts:98]: These refactorings are really nice

File: bigquery.ts (hunk 1 above)
Comment (maxf) [reply bigquery.ts:167]: Not sure, because the `results` is set differently in the `default` clause.
I'm aware that this could be written better though.

**Review by maxf — APPROVED** [review]
written and tested together

**Pull request merged**